File: Python/Programs/factorial.py
# Python program to find factorial of a number

num = int(input("Enter a number: "))
fact = 1
for i in range(1,num + 1):
    fact = fact * i
print(f"{num}! = {fact}")


"""
# short code

def fact(n):
    return n * fact(n - 1) if n > 1 else 1 # recusrsive and one-liner for finding factorial of argument   
print("5! = ", fact(5)) # sample

"""

File: C/Programs/Factorial.c
//C program to find factorial of a number

#include<stdio.h>
int main(){
    int num, i, factorial=1;
    printf("Enter a number: ");
    scanf("%d", &num); //getting number

    //logic for factorial
    for(i=1; i<=num; i++){
    factorial = factorial * i;
    }

    printf("%d! = %d", num, factorial); //printing factorial of the number
    return 0;
}